### PR TITLE
refactor: use summary details elements for disclosure

### PR DESCRIPTION
**Pull request by BigAB**
This PR removes some custom JS and CSS for showing and hiding the Liability Waiver and Member and Respect Guidelines, in favour of the built-in no-js `<details>` and `<summary>` elements, wrapped up in a `Disclosure` component for styles and text swapping of the open and closed state (which does not work without JS).

Note: This change actually ends up removing the animation, but hopefully only temporarily as it does include standards based CSS to animate the `details/summary` elements, but they are not supported in browsers yet.  That being said the animation does work in Chrome Canary and will "just start working" once the new CSS is supported in the users browser.

Chrome Canary:

https://github.com/user-attachments/assets/381eb677-8e63-43f0-ad90-58b4f1e87be3

### CODE REVIEW

File: src/lib/ui/Disclosure.svelte [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		block-size: calc-size(any, 0px);
+		transition-property: block-size, content-visibility;
+		transition-duration: 0.5s;
+		transition-behavior: allow-discrete;
+	}
+
+	details[open] > .content,
+	details[open]::details-content {
+		block-size: auto;
+	}
```
Comment (BigAB) [inline src/lib/ui/Disclosure.svelte:85]: This CSS is not supported in all browsers yet, but will be "soon", and the fallback is just not animating open and closed

File: src/lib/ui/Disclosure.svelte [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	{/if}
+	{#if $$slots['open-title'] && open}
+		<summary>
+			<slot name="open-title"></slot>
+		</summary>
+	{/if}
+	{#if $$slots['closed-title'] && !open}
+		<summary>
+			<slot name="closed-title"></slot>
+		</summary>
```
[the remark this answers is not in the document]
Comment (mikemitchel) [reply src/lib/ui/Disclosure.svelte:20]: 👍

File: src/lib/ui/Disclosure.svelte (hunk 1 above)
Comment (mikemitchel) [reply src/lib/ui/Disclosure.svelte:85]: future proof, i dig it

**Review by mikemitchel — APPROVED** [review]
read comments and understood, ship it

**Pull request merged**